websocket.go

```diff
@@ -0,0 +1,82 @@
+package goproxy
+
+import (
+	"bufio"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func headerContains(header http.Header, name string, value string) bool {
+	for _, v := range header[name] {
+		for _, s := range strings.Split(v, ",") {
+			if strings.EqualFold(value, strings.TrimSpace(s)) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func isWebSocketRequest(r *http.Request) bool {
+	return headerContains(r.Header, "Connection", "upgrade") &&
+		headerContains(r.Header, "Upgrade", "websocket")
+}
+
+func (proxy *ProxyHttpServer) handleWebsocket(ctx *ProxyCtx, tlsConfig *tls.Config, w http.ResponseWriter, req *http.Request, clientCon *tls.Conn) {
+	// Assuming wss since we got here from a CONNECT
+	targetURL := url.URL{Scheme: "wss", Host: req.URL.Host, Path: req.URL.Path}
+
+	// Run request through handlers
+	req, resp := proxy.filterRequest(req, ctx)
+	if resp != nil {
+		//TODO handle this
```

Comment:
`resp` is only not nil here if a handler has explicitly stopped the request from continuing, it should really be returned back to the client here instead of panicing or something. I'll make an issue